plugin/storage/factory.go

```diff
@@ -118,9 +118,15 @@ func (f *Factory) CreateSpanWriter() (spanstore.Writer, error) {
 		writers = append(writers, writer)
 	}
 	if len(f.SpanWriterTypes) == 1 {
```

Comment:
a) you can DRY it by first defining `topWrite` as either `writers[0]` or new sequential, and only then applying downsampler
b) shouldn't downsampler apply conditionally depending on the configuration (DownSamplingRatio)?

Reply:
a) will refactor
b) it won't apply if the ratio passed is 1, this is decided within the DownSamplerWriter. We will pass 1.0 as default ratio which means no down sampling at all.

Reply:
also, if we want to dynamically adjust the sampling rate, it's better to always create a DownSamplerWriter and handle the logic there.